addons/website_sale/views/templates.xml

```diff
@@ -2282,20 +2282,24 @@
                 <t t-else="">
                     <!-- No need to set href for address template as save address result will redirect
                     to proper place if all values are properly filled -->
-                    <t
-                        t-if="xmlid == 'website_sale.address'"
-                        t-set="main_button_href"
-                        t-value="False"
-                    />
-                    <t
-                        t-else=""
-                        t-set="main_button_href"
-                        t-value="step_specific_values['main_button_href']"
-                    />
+                    <t t-if="xmlid == 'website_sale.address'">
+                        <t t-set="main_button_href" t-value="False"/>
+                        <t
+                            t-set="main_button_label"
+                            t-value="'Continue checkout' if is_anonymous_cart else 'Save address'"
+                        />
```

Comment:
FYI translations in attributes are not exported so it's better to always put them in the element. 
```suggestion
                        <t t-if="is_anonymous_cart" t-set="main_button_label">
                             Continue checkout
                        />
```